### ISSUE TITLE
group name="A.Ways" group name="B.Traffic signs (by countries)"

### ISSUE THREAD

**Issue by d1g** [yopaseopor/traffic_signs_preset_JOSM#13]
### First group

 can be removed because it is almost equivalent to internal JOSM preset.

We have Planet-wide agreement about highway=* tags: http://wiki.openstreetmap.org/wiki/Proposed_features/Highway_key_voting_importance

Same about surface=* tag (not a country-specific tag) or tracktype=*

### Second group

Could be kept per every country. Thank you for the effort.

What resources were used by the way?

**Comment by yopaseopor** [comment]
I explain myself (sorry for my bad English)
This preset was born because of the need of a easy way to tag in a complete
style (more tags than only highway,oneway and ref) a way, with only few
clicks, some kind of wizard. At first this preset was only Spanish (for
Spain). So I made two sections: A. Vías (ways).

The intention was also a good tool, one-click preset, so we can complete a
way with necessary tags in a fast way you can't found in a way, also
surface for the tracks (and tracktypes, in Catalan community we have some
firefighters working with OSM and it was a way to make easier their job of
tagging tracks), oneway, noname (for the quality campaign 1carrer1nom
-1street1name-)

But also In Spain the situation is not easy as tag highway can show you.You
will think the more important way is better the way it is.No, amigo.Spain
is different. Here it works: the more money the local, regional or whatever
administration has, better the road it is. It is sad, but some Catalan -
one of the reasons of "independence Catalan process is this- or Basque or
Galician roads (autonomy government responsible)  are better than National
Spanish roads. So in Spain we have an "electric" discussion about that (you
can see Spanish talk-es list mails like :

https://lists.openstreetmap.org/pipermail/talk-es/2016-May/014001.html (in
Spanish)
<https://lists.openstreetmap.org/pipermail/talk-es/2016-May/014001.html>
or
https://lists.openstreetmap.org/pipermail/talk-es/2016-May/014014.html (in
Spanish)
or
http://yopaseopor.blogspot.com.es/2016/04/yomapeo-opinion-sobre-el-cambio-de.html
(in Spanish with pics)
and
http://yopaseopor.blogspot.com.es/2016/05/yomapeo-opinion-sobre-el-cambio-de.html
(in Spanish with pics)

So for this reason Spanish importance roads are not the same as highways
tags. You can see the OSM wiki Normalization Spanish page
https://wiki.openstreetmap.org/wiki/Normalizaci%C3%B3n .So because the
utility of the preset in Spain we have this group in all the presets (I
think in a future presets options will be adapted to each country with each
pic of different refs, as Spanish now is).

 B. Traffic signs (the complete set of 300 traffic signs set up in Spanish
traffic laws (BOE).

As the preset grew up and finished Spanish traffic signs I was thinking
what about other countries, and I started to search in taginfo, osm wiki
and thought it would be a good idea to add...other European countries
traffic signs? Why not?
I started with France, because is the country near Spain...but it was so
hard to translate code one by one...so first I had to organize that. So I
thought again...other European countries like Belgium, The Nederlands...
And started to consult the different European traffic laws to search for
the code of every traffic sign...First I have made Belgium because is the
lowest amount: 90+ I remember.Then I see Taginfo for search the most used:
It was not onlyGermany, it was The Nederlands or Finland. But it was so
hard and loong loong time to translate every code in every position.I don't
know anything about programming so...I try do the best I can when I can )

Also there was other dilemma, if traffic signs code are not unified in OSM
(only a few: city_limit,stop,give_way,maxheight,maxaxleload,masweight,maxspeed)
which preset is better: a preset where you can choose the traffic sign and
then choose the country (to had the code and style could identify it -
possibility of use all the Spanish code and order to make the preset) or
first you choose the country and then the traffic sign (possibility to add
your country to toolbar and only use that country)

I have chosen the country option. I was personally interested only in
Spanish traffic signs.
I have finished Spanish, Belgium and Dutch traffic signs...wait a minute ,
with only some Eurpean countries...what about others big countries...like
the USA?? What kind of traffic signs do they use - In overpass I see an
Spanish traffic sign code in Texas -  What law is? What system? I returned
to search in wikipedia's, osm wiki, and so one and found the MUTCD system.

Other problem: this preset is for JOSM. I am not a programmer so I make a
lot of mistakes that JOSM test doesn't like. The builders of JOSM tell me,
Hey man! your preset is breaking our tests...and our program, you can make
that or that, hey there are some mistakes you have to fix (I will thank
them all my life for their help). And the biggest problem: Now the preset
is bigger than JOSM itself.You have to split it. But I f I have to split
it...Will I have to do 25 presets???? Really now 29.
A big error in my JOSM out of memory made me to split the preset. So now
 every preset is a "structure clone" of each other: with ways and traffic
signs.

As finishing all the presets will be a big work for the years (in taginfo
https://taginfo.openstreetmap.org/projects/traffic_signs#tags you can see
25315 different options) with the help of Catalan Community I have found a
way to "build" an emergency option to show all the different kinds of
traffic signs per country with the only option to locate it in the way
(direction forward, backward, and also side: right,left,up,down).
Spanish preset is finished with all the options (you have also the option
to customize confirmation/orientation/destination traffic signs) and in
future I will port this option to other countries with their traffic signs.
Also the presets works with traffic_signs style and with JOSM's Kendzi3D
plug-in

The basic resources are: traffic laws of every country and Wikipedia and
OSM wiki with pages like http://wiki.openstreetmap.org/wiki/Key:traffic_sign.
My intention is to add...ALL the traffic signs of the ENTIRE WORLD.

The preset(s) is in construction, do you want to add something? some
country? translate it? Help would be appreciated.
Salut i senyals de trànsit
yopaseopor

**Comment by d1g** [comment]
@yopaseopor, quick reply about highway=* in Spain

https://wiki.openstreetmap.org/w/index.php?title=Normalizaci%C3%B3n&direction=next&oldid=92268 states about different approach since 13 April 2008
> Please note that most of the times we follow the **administrative classification** and not the actual road conditions.

We agreed to follow following highway=* definition in 2010: "**relative** importance in the **graph**":
> importance for the road grid (hierarchical position in the interconnecting network) instead of physical attributes.

This means: 1. independent from surface condition 

#### Extended Russian definition

In 2010 for Russia, we also [agreed](http://wiki.openstreetmap.org/wiki/RU:Highway_classification) to explicitly extend this to: 2. independent from local authorities and their claims - because their claims of "status" are often not usable in routing software or to recognize _practical road importance_.

Basically we picked top cities, and decided when they should be connected using specific hw=* class; classes were picked  _based on our local traffic knowledge_ and observations (not status at sign or on paper) and tried to explain it using "definitions" at wiki once.

2010 Russian highway=* definition means that you can have a road unrecognised by authorities, with terrible surface, but used by locals (5000-50000) as tertiary-secondary.
